gui/wxpython/psmap/frame.py

```diff
@@ -1980,9 +1980,19 @@ def RecalculatePosition(self, ids):
                 self.instruction[id]['rect'] = newRect
 
                 point1 = wx.Point2D(
-                    xDiff, yDiff) + self.instruction[id]['where'][0]
+                    *tuple(
+                        map(lambda i, j: i + j,
+                            (xDiff, yDiff),
+                            self.instruction[id]['where'][0])
+                    ),
+                )
                 point2 = wx.Point2D(
-                    xDiff, yDiff) + self.instruction[id]['where'][1]
+                    *tuple(
+                        map(lambda i, j: i + j,
+                            (xDiff, yDiff),
+                            self.instruction[id]['where'][1])
+                    ),
+                )
```

Comment:
Thanks for catching this, I think something like this would be more succinct. Also I think using list comprehension is in general considered preferable to `map`.

```
                 yDiff = newRect[1] - oldRect[1]
                 self.instruction[id]['rect'] = newRect
 
-                point1 = wx.Point2D(
-                    xDiff, yDiff) + self.instruction[id]['where'][0]
-                point2 = wx.Point2D(
-                    xDiff, yDiff) + self.instruction[id]['where'][1]
+                point1 = wx.Point2D(self.instruction[id]['where'][0])
+                point2 = wx.Point2D(self.instruction[id]['where'][1])
+                point1 += wx.Point2D(xDiff, yDiff)
+                point2 += wx.Point2D(xDiff, yDiff)
                 self.instruction[id]['where'] = [point1, point2]
 
                 self.RecalculateEN()

```

Reply:
Yes, you're right.